[limeiy1] iP

limeiy1 opened the pull request.

limeiy1 committed:
rename, greet, exit

limeiy1 committed:
Rename

limeiy1 committed:
Echo

limeiy1 committed:
Add Task class

limeiy1 committed:
Add, List

Add the ability to store whatever text entered by the user and display them back to the user when requested.

limeiy1 committed:
complete Task class

getStatusIcon(), markAsDone(), unmarkAsDone(), toString()

limeiy1 committed:
update Duke to include mark as done function

added ability to mark and unmark the tasks as done

duckyfuz commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,59 @@
+            line = in.nextLine();
+            if (line.equalsIgnoreCase("bye")) {
+                System.out.println(LINE);
+                System.out.println("Bye. Hope to see you again soon!");
+                System.out.println(LINE);
+                return;
+            }
+            if (line.equalsIgnoreCase("list")) { // show list
+                System.out.println(LINE);
+                if (count == 0) {
+                    System.out.println("List is empty!");
+                } else {
+                    System.out.println("Here are the tasks in your list:");
+                    for (int i = 0; i < count; i++) {
+                        System.out.println((i + 1) + ". " + list[i].toString());
+                    }
+                }
+                System.out.println(LINE);
+            } else if (line.startsWith("mark")) { // mark as done
+                int taskIndex = Integer.parseInt(line.split(" ")[1]) - 1;
+                list[taskIndex].markAsDone();
+                System.out.println(LINE);
+                System.out.println("Nice! I've marked this task as done:");
+                System.out.println(list[taskIndex].toString());
+                System.out.println(LINE);
+            } else if (line.startsWith("unmark")) { // unmark as done
+                int taskIndex = Integer.parseInt(line.split(" ")[1]) - 1;
+                list[taskIndex].unmarkAsDone();
+                System.out.println(LINE);
+                System.out.println("OK, I've marked this task as not done yet:");
+                System.out.println(list[taskIndex].toString());
+                System.out.println(LINE);
+            } else { // add new task
+                System.out.println(LINE);
+                System.out.println("added: " + line);
+                System.out.println(LINE);
+                list[count] = new Task(line);
+                count++;
+            }
+        } while (in.hasNextLine());
```
I like the organization of the main method with its use of clear separation between the setup of messages and the command handling loop. It makes following the program flow easier.

duckyfuz commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,59 @@
+        System.out.println(LINE);
+
+        Scanner in = new Scanner(System.in);
+        String line;
+        Task[] list = new Task[100];
+        int count = 0;
+        do {
+            line = in.nextLine();
```
Consider using more descriptive variable names than line, count and list. For example, commandLine, taskCount, and tasksList could make the code more self-explanatory.

duckyfuz commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,59 @@
+        int count = 0;
+        do {
+            line = in.nextLine();
+            if (line.equalsIgnoreCase("bye")) {
+                System.out.println(LINE);
+                System.out.println("Bye. Hope to see you again soon!");
+                System.out.println(LINE);
+                return;
+            }
+            if (line.equalsIgnoreCase("list")) { // show list
+                System.out.println(LINE);
+                if (count == 0) {
+                    System.out.println("List is empty!");
+                } else {
+                    System.out.println("Here are the tasks in your list:");
+                    for (int i = 0; i < count; i++) {
+                        System.out.println((i + 1) + ". " + list[i].toString());
+                    }
+                }
+                System.out.println(LINE);
+            } else if (line.startsWith("mark")) { // mark as done
+                int taskIndex = Integer.parseInt(line.split(" ")[1]) - 1;
+                list[taskIndex].markAsDone();
+                System.out.println(LINE);
+                System.out.println("Nice! I've marked this task as done:");
+                System.out.println(list[taskIndex].toString());
+                System.out.println(LINE);
+            } else if (line.startsWith("unmark")) { // unmark as done
```
Perhaps we can define constants like "list", "bye", "mark", and "unmark" so that we no longer need to hard-code these words within the logic. Can loop up magic strings.

duckyfuz commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,25 @@
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    public void markAsDone() {
+        isDone = true;
+    }
+
+    public void unmarkAsDone() {
+        isDone = false;
+    }
```
The methods markAsDone() and unmarkAsDone() are well named imo! Intuitive and describes the method well.

duckyfuz reviewed (commented):
The overall structure and sequence of operations are logically presented. I think care was taken to clarify user interaction and task management, which makes it easier for others to follow your logic.

limeiy1 committed:
Create Todo, Deadline, and Event classes

limeiy1 committed:
update code to track three types of tasks:

1. ToDos: tasks without any date/time attached to it e.g., visit new theme park
2. Deadlines: tasks that need to be done before a specific date/time e.g., submit report by 11/10/2019 5pm
3. Events: tasks that start at a specific date/time and ends at a specific date/time e.g., (a) team project meeting 2/10/2019 2-4pm (b) orientation week 4/10/2019 to 11/10/2019

limeiy1 committed:
implement automated text UI testing

limeiy1 committed:
rename count to numberOfTasks, get rid of magic number by using MAX_TASKS

limeiy1 committed:
improve code quality:
eg rename variables, reduce deep nesting

limeiy1 committed:
create DukeException

limeiy1 committed:
handle errors

limeiy1 committed:
handle event errors

limeiy1 committed:
update ui test to test error handling

limeiy1 committed:
Merge branch 'branch-Level-5'

limeiy1 committed:
rename class to Mimi

limeiy1 committed:
Organise classes into java packages

limeiy1 committed:
update ui test

limeiy1 committed:
Merge branch 'branch-A-Packages'

Charly2312 commented on src/main/java/mimi/Mimi.java:
```diff
@@ -0,0 +1,172 @@
+    private static int taskCount = 0;
+    static Task[] taskList = new Task[MAX_TASKS];
+
+    public static void main(String[] args) {
```
Hi, overall this looks good but just a small detail, try to keep each function at 30 lines max. So just small detail, for the "hello" and "what can I do.." maybe can abstract them into a function

Charly2312 commented on src/main/java/mimi/Mimi.java:
```diff
@@ -0,0 +1,172 @@
+                } else if (inputLine.startsWith("event")) { // add new event task
+                    addEvent(inputLine);
+                } else { // print a statement when an invalid command is received
+                    throw new MimiException("Please enter a valid command :(");
```
For this one, just a small detail, maybe can make into a constant like VALID_COMMAND: "Please enter .." . This applies to other error messages or similar statements

Charly2312 commented on src/main/java/mimi/Mimi.java:
```diff
@@ -0,0 +1,172 @@
+            "____________________________________________________________";
+    private static final int MAX_TASKS = 100;
+    private static int taskCount = 0;
+    static Task[] taskList = new Task[MAX_TASKS];
```
just keep in mind, for week 6, you will learn about ArrayList. can consider using that

Charly2312 commented on src/main/java/mimi/Mimi.java:
```diff
@@ -0,0 +1,172 @@
+        System.out.println(LINE);
+    }
+
+    private static void addTodo(String line) throws MimiException {
```
good use of exception, keep it up

Charly2312 commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,32 @@
-|____/ \__,_|_|\_\___|
-
+____________________________________________________________
+Hello I'm mimi!
```
good attempt in trying to to do the txt test. Hope it works

Charly2312 commented on src/main/java/mimi/Mimi.java:
```diff
@@ -0,0 +1,172 @@
+                    System.out.println(LINE);
+                    return;
+                }
+                if (inputLine.equalsIgnoreCase("list")) { // show list
```
maybe can extract these cases into a function like handleActionInputs or something like that

Charly2312 reviewed (commented):
overall has very good code quality and readability. Keep it up!

limeiy1 committed:
add gitignore file

limeiy1 committed:
implement ArrayList

limeiy1 committed:
Add support for deleting tasks from the list.

limeiy1 committed:
Save the tasks in automatically whenever the task list changes

limeiy1 committed:
Merge branch 'branch-Level-6'

limeiy1 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/mimi/Mimi.java

limeiy1 committed:
change if-else to switch, add more error handling

limeiy1 committed:
add more classes

limeiy1 committed:
organise into packages

limeiy1 committed:
add code that deals with making sense of the user command

limeiy1 committed:
add code that deals with interactions with the user

limeiy1 committed:
add code in Storage to deal with loading tasks from the file and saving tasks in the file
add code in TaskList which contains the task list, and has operations to add/delete tasks in the list
add command classes

limeiy1 committed:
update rest of code using new classes

limeiy1 committed:
add FindCommand
update code to give users a way to find a task by searching for a keyword in the task description

limeiy1 committed:
Merge pull request #1 from limeiy1/branch-Level-9

add FindCommand

limeiy1 committed:
Merge branch 'master' of https://github.com/limeiy1/ip

* 'master' of https://github.com/limeiy1/ip:
  add FindCommand update code to give users a way to find a task by searching for a keyword in the task description

limeiy1 committed:
Add JavaDoc comments to the code. Add header comments to non-private classes/methods.

limeiy1 committed:
Merge pull request #2 from limeiy1/branch-A-JavaDoc

Add JavaDoc comments to the code. Add header comments to non-private …

limeiy1 committed:
Merge branch 'master' of https://github.com/limeiy1/ip

* 'master' of https://github.com/limeiy1/ip:
  Add JavaDoc comments to the code. Add header comments to non-private classes/methods.

limeiy1 committed:
update README with features list

limeiy1 committed:
update intro of README

limeiy1 committed:
Add Quick Start to README

limeiy1 committed:
README update

limeiy1 committed:
update README

limeiy1 committed:
update README content page

limeiy1 committed:
README update

limeiy1 committed:
add notes about command format to README

limeiy1 committed:
update README links

limeiy1 committed:
Update ui-test

limeiy1 committed:
Add link to jar file in README.md
